server/src/main/java/org/elasticsearch/action/admin/indices/rollover/MetadataRolloverService.java

```diff
@@ -161,10 +161,9 @@ static CreateIndexClusterStateUpdateRequest prepareCreateIndexRequest(final Stri
      * the rollover alias will point to multiple indices. This causes indexing requests to be rejected.
      * To avoid this, we make sure that there is no duplicated alias in index templates before creating a new index.
      */
-    static void checkNoDuplicatedAliasInIndexTemplate(
-            Metadata metadata, String rolloverIndexName, String rolloverRequestAlias,
-            @Nullable Boolean isHidden) {
-        final List<IndexTemplateMetadata> matchedTemplates = findTemplates(metadata, rolloverIndexName, isHidden);
+    static void checkNoDuplicatedAliasInIndexTemplate(Metadata metadata, String rolloverIndexName, String rolloverRequestAlias,
+                                                      @Nullable Boolean isHidden) {
+        final List<IndexTemplateMetadata> matchedTemplates = findV1Templates(metadata, rolloverIndexName, isHidden);
```

Comment:
Should we check v2 templates here too? If so, then I think it is ok to do this in a followup change as well.

Reply:
Definitely, I'll be addressing these in followups!